internal/storage/clickhouse.go

```diff
@@ -0,0 +1,91 @@
+package storage
+
+import (
+	"context"
+	"crypto/tls"
+	"fmt"
+	"os"
+	"strconv"
+	"time"
+
+	"github.com/ClickHouse/clickhouse-go/v2"
+	"github.com/ethereum/go-ethereum/log"
+)
+
+type ClickHouseConnector struct {
+	cache clickhouse.Conn
+}
+
+type ClickhouseConnectorConfig struct {
+	ExpiresAt time.Duration
+}
+
+func NewClickHouseConnector(cfg *ClickhouseConnectorConfig) (*ClickHouseConnector, error) {
+	conn, err := ConnectDB()
+	// Question: Should we add the table setup here?
```

Comment:
The connector doesn't have any knowledge of which tables to create though

Reply:
The tables are the same for each chain and the chain name is the name of the schema, so it should be douable